### PR TITLE
Cargo fmt + minor stuff

### LINKED ISSUE #28: Is `rustfmt` welcome here?
Currently the code base does not use `rustfmt`. As this is a somewhat contentious topic I thought I'd ask before opening a PR to enforce it.

### PR DESCRIPTION
**Pull request by maciejhirsz**
+ `cargo fmt` (fixes #28).
+ Used the `concat!` macro in response headers to make them a notch faster.
+ ~Added `body` to `Response::Reject` so the user can send intelligible errors back.~

### CODE REVIEW

**Review by dvdplm — APPROVED** [review]
LGTM.

I would prefer it if the changes to the rejection error to include the body were in a separate PR but I'm ok merging it here too if @jsdw and @niklasad1 are on-board as well.

**Comment by maciejhirsz** [comment]
Rolled back the body, it works, but it doesn't even show up in firefox dev tools, so it's kind of useless :)

**Comment by dvdplm** [comment]
One thing though: we have a semi-standard `rustfmt.toml` file (e.g. [here](https://github.com/paritytech/jsonrpsee/blob/9a02c10a311c36185b13d2d8d71d37ee44c16c0b/rustfmt.toml#L2)). I think we should use it here too.

So it'd be:

```toml
hard_tabs = true
max_width = 120
use_small_heuristics = "Max"
edition = "2018"
```

File: src/base.rs [hunk 1]
```diff
@@ -560,26 +565,23 @@ pub enum Error {
...
+            Error::UnknownOpCode => f.write_str("unknown opcode"),
+            Error::ReservedOpCode => f.write_str("reserved opcode"),
+            Error::FragmentedControl => f.write_str("fragmented control frame"),
+            Error::InvalidControlFrameLen => f.write_str("invalid control frame length"),
+            Error::InvalidReservedBit(n) => write!(f, "invalid reserved bit: {}", n),
+            Error::PayloadTooLarge { actual, maximum } => write!(
+                f,
+                "payload too large: len = {}, maximum = {}",
+                actual, maximum
+            ),
```
Comment (jsdw) [inline src/base.rs]: That's a shame; I liked how this had been laid out :)

**Comment by maciejhirsz** [comment]
> One thing though: we have a semi-standard `rustfmt.toml` file (e.g. [here](https://github.com/paritytech/jsonrpsee/blob/9a02c10a311c36185b13d2d8d71d37ee44c16c0b/rustfmt.toml#L2)). I think we should use it here too.

Applied!

**Review by niklasad1 — APPROVED** [review]

**Review by jsdw — APPROVED** [review]

**Pull request merged**